pilot/platform/kube/inject/mesh.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inject
+
+import (
+	"fmt"
+	"html/template"
+	"os"
+)
+
+type Init struct {
+	Image      string
+	Args       []string
+	PullPolicy string
+}
+
+type Container struct {
+	Image      string
+	Args       []string
+	PullPolicy string
+	Env        []string
+}
+
+type MeshConfig struct {
+	Init
+	Container
+}
+
+var (
+	serviceEscape = `
+{{ if eq .ServiceCluster "" }}
+- {{ "istio-proxy" }}
+{{ else }}
+- {{ printf "%s" .ServiceCluster }}
+{{ end }}`
+	meshTemplate = `
+initContainers:
```

Comment:
@ayj Should all of these things just reuse the types from proxy.MeshConfig and proxy.DefaultConfig ? I'd like to have these things strongly typed, but I'm not sure if embedding those will cause a lot of confusion?

Reply:
cc @kyessenov

Reply:
I would do whatever is simplest to get the lower-level templated sidecar injection configuration merged. I suspect this will be using `proxy.MeshConfig` and `proxy.DefaultConfig` directly. The higher-level template generation process, whether through helm (https://github.com/istio/istio/pull/2030), or the multi-cluster config (https://github.com/istio/istio/issues/1980) can be refactored on-top.

Reply:
Yeah, doing something simple is probably the right thing here. I suspect the mesh config resource will go through another round of revision at some point soon.